internal/persistence/json/database.go

```diff
@@ -79,16 +83,32 @@ var (
 	ErrIPRecordsMisordered = errors.New("IP records are not ordered correctly by time")
 	ErrIPEmpty             = errors.New("IP is empty")
 	ErrIPTimeEmpty         = errors.New("time of IP is empty")
+	ErrVersionMismatch     = errors.New("invalid database version")
 )
 
 func checkData(data dataModel) error {
+	var version int
+	switch data.Version {
+	case "":
+		version = 0
+	case "1":
+		version = 1
+	default:
+		return fmt.Errorf("%w: %q", ErrVersionMismatch, data.Version)
```

Comment:
nit using `%s` is fine, we don't really need to surround it with double quotes 😉 

```suggestion
		return fmt.Errorf("%w: %s", ErrVersionMismatch, data.Version)
```